### DISCUSSION TITLE
Question

### DISCUSSION THREAD

**Discussion by bameyrick** [ONSdigital/design-system#1393] [Feedback]
Use this place to discuss the [question pattern](https://ons-design-system.netlify.com/patterns/question) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by bameyrick** [comment]
Classes parameter does not work as its not just `classes` in the macro template, not `params.classes`

**Reply by jrbarnes9** [reply]
Issue #381 resolved

**Comment by bameyrick** [comment]
When the question component is in fieldset mode and there is no question description, the spacing between the title and the legend is too big.

This is because the legend and the h1 both have a margin bottom.

![Screenshot 2019-05-17 at 11 26 31](https://user-images.githubusercontent.com/8447446/57922035-bd6d5f80-7896-11e9-93a8-af7279c32129.png)

**Reply by jrbarnes9** [reply]
Issue #383 resolved

**Comment by bameyrick** [comment]
Question margins are still wrong, when there is a `h1` wrapped in a `legend` followed by `p` tags, the `p` tags aren't effected by the `h1`'s margin bottom:

![Image 28-05-2019 at 14 54](https://user-images.githubusercontent.com/8447446/58483833-ddcfd080-8158-11e9-8667-c1ec06275745.jpg)

**Reply by jrbarnes9** [reply]
Issue #394 resolved

**Comment by bameyrick** [comment]
When the `h1` is within the `legend`,  JAWS fails to find the `h1` when navigating by headings.

**Reply by jrbarnes9** [reply]
Issue #450 resolved

**Comment by CFryer** [comment]
Number of broken links encountered:

https://ons-design-system.netlify.com/components/radio
https://ons-design-system.netlify.com/components/date
https://ons-design-system.netlify.com/pattern/durations

**Comment by jrbarnes9** [comment]
Ensure consistent vertical spacing
Issues include:
- Vertical spacing is sometimes defined varies between top and bottom margins
- Spacing between a question page button varies depending on last item in question block

**Reply by jrbarnes9** [reply]
Issue #824 raised

**Comment by jrbarnes9** [comment]
Update previous links in question page pattern to use breadcrumbs component

**Reply by jrbarnes9** [reply]
Issue #1523 resolved

**Comment by jrbarnes9** [comment]
Question descriptions produce invalid html when a `p` tag is used within them 

...because they are marked up with `span`

**Reply by jrbarnes9** [reply]
Issue #1673 raised

**Comment by jrbarnes9** [comment]
`SubmitButton` param not documented in "anatomy" question pattern example

**Reply by jrbarnes9** [reply]
Raised #1910
